pkg/build/webhook/generic/generic.go

```diff
@@ -28,7 +29,7 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		return
 	}
 	glog.V(4).Infof("Checking if the provided secret for BuildConfig %s/%s matches", buildCfg.Namespace, buildCfg.Name)
-	if trigger.GenericWebHook.Secret != secret {
+	if !subtle.ConstantTimeCompare([]byte(trigger.GenericWebHook.Secret), []byte(secret)) {
```

Comment:
`ConstantTimeCompare` returns an integer. You can use `hmac.BytesEqual` (from crypto/hmac) which does the same thing (actually it uses `ConstantTimeCompare` under the hood) but returns a boolean aka it's more readable

https://golang.org/src/crypto/hmac/hmac.go?s=2437:2471#L87